hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/logical/ValuesUnionLogicalRule.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl.opt.logical;
+
+import com.google.common.collect.ImmutableList;
+import com.hazelcast.jet.sql.impl.opt.OptUtils;
+import org.apache.calcite.plan.RelOptRule;
+import org.apache.calcite.plan.RelOptRuleCall;
+import org.apache.calcite.plan.RelOptRuleOperand;
+import org.apache.calcite.rel.core.RelFactories;
+import org.apache.calcite.rel.core.Union;
+import org.apache.calcite.rel.logical.LogicalUnion;
+import org.apache.calcite.rel.logical.LogicalValues;
+import org.apache.calcite.rex.RexLiteral;
+
+import static java.util.stream.Collectors.collectingAndThen;
+import static java.util.stream.Collectors.toList;
+
+final class ValuesUnionLogicalRule extends RelOptRule {
+
+    static final RelOptRule INSTANCE;
+
+    private static final RelOptRuleOperand CHILD_OPERAND;
+
+    static {
+        CHILD_OPERAND = operand(LogicalValues.class, none());
+        INSTANCE = new ValuesUnionLogicalRule();
+    }
+
+    private ValuesUnionLogicalRule() {
```

Comment:
What is the purpose of this rule? AFAIU it is used to simplify `(values UNION values)` or alike, but at the same time there are no tests, and the `UNION` keyword is not allowed in the `UnsupportedOperationVisitor`. Looks like a dead code to me.

Reply:
Queries like `SELECT * FROM (VALUES (1), (1 + 2))` (with an expression in one of the rows) are translated to `UNION` of `VALUE`s by Calcite. For such cases we have tests.

Reply:
It also allows `VALUES (1), (2)`, that is multiple rows in `VALUES`. See for example [here](https://github.com/viliam-durina/hazelcast-jet/blob/4ec97b46b184292898a3a33f417c38fa3ce549fa/hazelcast-jet-sql/src/test/java/com/hazelcast/jet/sql/SqlFilterProjectTest.java#L53).

Reply:
However, `VALUES (1), (2)` does not need support from `ValuesUnionLogicalRule`.

Reply:
I see, thanks.